app/assets/javascripts/components/modals/confirmModal.jsx

```diff
@@ -0,0 +1,56 @@
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import { actions as appActions } from 'redux/modules/app/index.js';
+import Modal from 'react-modal';
+import SignalIcon from 'components/links/signal_icon.jsx';
+
+
+class ConfirmModal extends Component {
+  constructor(props) {
+    super(props);
+    this.closeModal = this.closeModal.bind(this);
+    this.onConfirm = this.onConfirm.bind(this);
+  }
+
+  componentWillMount() {
+    Modal.setAppElement('body');
+  }
+
+  onConfirm() {
```

Comment:
It's not returning back a promise because you're not returning one. I'll write a comment above as to where you should be returning.